src/paths/distances.c

```diff
@@ -1049,3 +1052,131 @@ igraph_error_t igraph_graph_center_dijkstra(
 
     return IGRAPH_SUCCESS;
 }
+
+// helper functions for now
+// TODO: integrate somewhere?
+static void update_to_min(igraph_integer_t *a, igraph_integer_t b) {*a = (*a > b) ? b : *a;}
+static void update_to_max(igraph_integer_t *a, igraph_integer_t b) {*a = (*a > b) ? *a : b;}
+
+/**
+ * \function igraph_diameter_bound
+ * TODO: update
+ * Plan: return diameter
+ */
+igraph_error_t igraph_diameter_bound(
+    const igraph_t *graph,  // input graph
+    igraph_integer_t *diameter,  // output diameter value
+    igraph_integer_t vid_start,  // vertex to start search from, if negative we choose
+    igraph_integer_t *from,  // output start of longest shortest path
+    igraph_integer_t *to,  // output end of longest shortest path
+    igraph_bool_t directed,  // is this graph directed?
+    igraph_bool_t unconn  // is this graph connected
+) {
+    directed = 0; unconn = 0;  // TODO: remove
+    if (directed || unconn) {
+        // for now, don't allow these cases // TODO: fix
+        printf("%d %d\n", directed, unconn);
+        IGRAPH_ERROR("Don't support this yet", IGRAPH_EINVAL);
+    }
+
+    igraph_integer_t no_of_nodes = igraph_vcount(graph);
+    if (vid_start >= no_of_nodes) {
+        IGRAPH_ERROR("Starting vertex ID for pseudo-diameter out of range.", IGRAPH_EINVAL);
+    }
+
+    printf("start: %li\n", vid_start);
+    if (vid_start < 0) {
+        // no given start vertex, so we choose one ourselves
+        igraph_maxdegree_arg(graph, &vid_start, igraph_vss_all(), IGRAPH_ALL, IGRAPH_LOOPS);
+        // TODO: not sure about this vids selector
+        // TODO: something about loop being slower?
+        printf("start: %li\n", vid_start);
+    }
+
+    // initialise set W (line 3)
```

Comment:
Which part of the paper does this refer to?

Reply:
Discussed offline; these refer to the "Determining the Diameter of Small World Networks" paper.